workspaces/adventure-pack/goodies/typescript/Iterator.prototype.toSet/index.ts

```diff
@@ -0,0 +1,12 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    toSet(this: Iterator<T>): Set<T>;
+  }
+}
+
+iteratorPrototype.toSet ??= function <T>(this: Iterator<T>): Set<T> {
+  return new Set([...this.toIterable()]);
```

Comment:
The [Set constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Set) should accept any iterable, so we can avoid the intermediate array!

```suggestion
  return new Set(this.toIterable());
```